### ISSUE TITLE
Data decrease in protein_pdb_map datasets

### ISSUE THREAD

**Issue by katewarner** [glygener/glygen-issues#1815]
During protein dataset QC I found that the number of UniProt ACs and rows in all the `protein_pdb_map.csv `datasets had decreased. I'm not sure why but I think that a major reason may be that they are missing the "experimental_method" AlphaFold data  

e.g. unreviewed/fruitfly_protein_pdb_map.csv

```
[k.warner1@glygen-vm-dev glygen]$ awk -F, '{print $1}' /data/shared/glygen/releases/data/v-2.6.1/reviewed/fruitfly_protein_pdb_map.csv | sort -u | grep -v "uniprotkb_canonical_ac" | wc -l
13246
[k.warner1@glygen-vm-dev glygen]$ awk -F, '{print $1}' /data/projects/glygen/generated/datasets/unreviewed/fruitfly_protein_pdb_map.csv | sort -u | grep -v "uniprotkb_canonical_ac" | wc -l
647
[k.warner1@glygen-vm-dev glygen]$ grep 'AlphaFold' /data/shared/glygen/releases/data/v-2.6.1/reviewed/fruitfly_protein_pdb_map.csv | wc -l
13139
[k.warner1@glygen-vm-dev glygen]$ grep 'AlphaFold' /data/projects/glygen/generated/datasets/unreviewed/fruitfly_protein_pdb_map.csv | wc -l
0
```

**Comment by rykahsay** [comment]
Please check now

**Comment by katewarner** [comment]
Looks good. Many thanks

**Comment by katewarner** [comment]
@jieluo321 I've noticed that we are missing a number of accessions with cross-references to "alphafolddb" in our pig, fruitfly and human nt files, that were present in our last release but are not present in the latest nt files  even though they have a Alphafold xref on the UniProt website e.g.  
```
[k.warner1@glygen-vm-dev 2024_06_20]$ cat uniprot-proteome-sus-scrofa.nt | grep alphafolddb/K7GRV3
<http://purl.uniprot.org/uniprot/K7GRV3> <http://www.w3.org/2000/01/rdf-schema#seeAlso> <http://purl.uniprot.org/alphafolddb/K7GRV3> .
<http://purl.uniprot.org/alphafolddb/K7GRV3> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://purl.uniprot.org/core/Resource> .
<http://purl.uniprot.org/alphafolddb/K7GRV3> <http://purl.uniprot.org/core/database> <http://purl.uniprot.org/database/AlphaFoldDB> .
[k.warner1@glygen-vm-dev 2024_06_20]$ cd /data/projects/glygen/downloads/ebi/current/
[k.warner1@glygen-vm-dev current]$ cat uniprot-proteome-sus-scrofa.nt | grep alphafolddb/K7GRV3
[k.warner1@glygen-vm-dev current]$ 
```
Have you removed a number of AlphaFold xrefs from these species which will be visible on the website after the next UniProt release, or is there an issue with these nt files?

For your information, I've pasted the accessions missing alphafolddb xrefs in the latest `uniprot-proteome-sus-scrofa.nt` file below: 

```
A0A287AA08-1
A0A5G2QFX2-1
F1RUG0-1
A0A287AEG6-1
I3LJG3-1
A0A287AVW9-1
A0A286ZJ05-1
A0A5G2QS75-1
A0A5G2QDV6-1
A0A287BN46-1
A0A5G2QHR3-1
A0A287A0L0-1
A0A287ABB6-1
I3LJT0-1
I3L7T1-1
A0A5G2QU73-1
I3L5Y9-1
F1SGG6-1
K7GPV8-1
F1RIW2-1
A0A287AZS7-1
F1RU69-1
F1S0V7-1
A0A287BSI0-1
A0A287BG14-1
F1S2F6-1
A0A287AMR3-1
I3LDI0-1
F1SIM2-1
F1RP47-1
A0A5G2QTK4-1
A0A287ANQ2-1
A0A286ZYU5-1
F1RUP0-1
A0A5G2R1S5-1
A0A287B916-1
A0A287BJ72-1
F1RFI0-1
A0A5G2QUV3-1
A0A5G2RMU7-1
K7GRV3-1
K7GMW9-1
A0A5G2R6Q8-1
A0A287A440-1
A0A287ART4-1
A0A5G2QE76-1
A0A5G2Q797-1
A0A287AKF8-1
A0A287AIN4-1
A0A286ZQK2-1
I3LGB1-1
A0A5G2R1P5-1
A0A286ZPJ4-1
A0A5G2QT19-1
A0A287AZK8-1
I3LT93-1
A0A286ZV71-1
A0A287AB27-1
A0A287BPB6-1
A0A5G2QTW4-1
A0A5G2RJP8-1
A0A287AAB5-1
F1S9F7-1
F1RX00-1
F1S418-1
I3L9E5-1
K7GKI5-1
```

**Comment by katewarner** [comment]
@jieluo321 Hi Jie, just wondering if you've had a chance to look into this?

**Comment by jieluo321** [comment]
I checked K7GRV3: https://rest.uniprot.org/uniprotkb/K7GRV3.txt there is no AlphaFoldDB cross references. AlphaFoldDB is not fully synchronised with uniprotkb entries. AlphaFoldDB is several release behind uniprotkb release. So some mismatch will always be there.

**Comment by katewarner** [comment]
@rykahsay as Jie said above, a number of alphafoldDB IDs have been removed from the latest fruitfly, human and pig `protein_pdb_map.csv` datasets, but that appears to be correct when I checked the UniProt api files. All the IDs removed from the latest Fruitfly dataset (20 IDs) no longer have an alphafoldDB xref in UniProt.  

As for the human and pig datasets, I ran the `check-dataset-group.py` script again and they are still missing a number of IDs, that have alphafoldDB xrefs in UniProt, and have alphafoldDB xrefs in their nt files. The accessions I checked are all `"unreviewed_isoforms"` in the organisms `protein_masterlist.csv` so I'm wondering if this might be the problem. 

Here are the ACs that should be in the `protein_pdb_map.csv` datasets.   

```
Missing ACs in human_protein_pdb_map.csv
E7EX29
A0A1B0GTE1
G3V2N0
A0A0G2JKD1
Q9H252
Q9Y4K1
E9PQZ5
E7ES20
P49761
E7EQ52
Q5VXM1
A0A0B4J1S8
Q9NVX7
B7WNX0

Missing ACs in pig_protein_pdb_map.csv
A0A287AA08
A0A5G2QFX2
A0A287AEG6
I3LJG3
A0A286ZJ05
A0A5G2QS75
A0A5G2QDV6
A0A287BN46
A0A287ABB6
I3L7T1
F1RIW2
A0A287AZS7
A0A287BG14
F1S2F6
I3LDI0
F1SIM2
F1RP47
A0A286ZYU5
A0A5G2R1S5
A0A287BJ72
F1RFI0
A0A5G2QUV3
A0A5G2RMU7
K7GMW9
A0A5G2R6Q8
A0A287ART4
A0A5G2QE76
A0A5G2Q797
A0A5G2R1P5
A0A286ZPJ4
A0A5G2QT19
A0A287AZK8
I3LT93
A0A286ZV71
A0A287AB27
A0A287AAB5
F1RX00
F1S418
```

**Comment by rykahsay** [comment]
I am confused, if I looked in your first example ("E7EX29"), it looks like it didn't have AlphaFold in 2.6.1 (see below) -- it only has PDB.

```
$ cat /data/shared/glygen/releases/data/v-2.6.1/reviewed/human_protein_pdb_map.csv | grep E7EX29
"E7EX29-1","region_1","8p1h","a/b","1","72","1.0","0.75","X-Ray_Crystallography","1.95","True"
"E7EX29-1","region_1","8p1h","a/b","76","164","1.0","0.75","X-Ray_Crystallography","1.95","False"
"E7EX29-1","region_1","8p1h","a/b","168","237","1.0","0.75","X-Ray_Crystallography","1.95","False"
```

The question should be, why are the above rows missing in unreviewed/human_protein_pdb_map.csv
```
$ cat unreviewed/human_protein_pdb_map.csv  | grep E7EX29
```

The answer is: because the current uniprot-proteome-homo-sapiens.nt does not have them.

```
$ cat downloads/ebi/current/uniprot-proteome-homo-sapiens.nt | grep " <http://purl.uniprot.org/core/chain> " | grep E7EX29
```

But the old nt file (downloads/ebi/2024_06_20/uniprot-proteome-homo-sapiens.nt ) has them
```
$ cat downloads/ebi/2024_06_20/uniprot-proteome-homo-sapiens.nt | grep " <http://purl.uniprot.org/core/chain> " | grep E7EX29
<http://purl.uniprot.org/isoforms/E7EX29-1#PDB_8P1H_tt1tt72> <http://purl.uniprot.org/core/chain> "A/B=1-72" .
<http://purl.uniprot.org/isoforms/E7EX29-1#PDB_8P1H_tt76tt164> <http://purl.uniprot.org/core/chain> "A/B=76-164" .
<http://purl.uniprot.org/isoforms/E7EX29-1#PDB_8P1H_tt168tt237> <http://purl.uniprot.org/core/chain> "A/B=168-237" .
```

**Comment by katewarner** [comment]
@rykahsay I'm looking further into this but want to check something. For the `protein_pdb_map` files do we only provide mapping for PDB and Alphafold xrefs for the canonical ACs? 

I've noticed that many of the IDs with Alphafold xrefs that have been removed in the latest release, are classed as "unreviewed_isoforms" in their `protein_masterlist` dataset.

**Comment by rykahsay** [comment]
"... do we only provide mapping for PDB and Alphafold xrefs for the canonical ACs?" 
It was not the case but I have now I have changed it like that.

I have done a comparison of with 2.6.1 considering only canonical accessions and rows with "selection_flag" = "True".

Have a look the files shown below  -- "additions.txt" contains canonical accessions that new in this release and "missing.txt" contains canonicals that were mapped to structure in 2.6.1 but are not in current pdb_map datasets (unreviewed/*_protein_pdb_map.csv)

```
 /data/projects/glygen/generated/datasets/logs/additions.txt
 /data/projects/glygen/generated/datasets/logs/missing.txt
```

**Comment by katewarner** [comment]
It looks like the data decrease in these datasets are expected and not an issue. I looked at the missing IDs in the human and pig datasets and it looks like they are either accessions in which UniProt has removed their Alphafold xrefs from their database or they are unreviewed accessions which have been removed from the latest pdb_map dataset because you now only map to the canonical accessions. This means the data decrease can be explained so I'm going to close this ticket. Many thanks for your help
